Extends overrides section of base config

arikon opened the pull request:
It was overwritten before

jugglinmike commented:
Thanks for the pull request! Could you add a test for this?

arikon commented:
@jugglinmike There are no tests for `extends` at all. Could you spot me, where to add it?

jugglinmike commented:
https://github.com/jshint/jshint/blob/master/tests/cli.js . That file [currently uses `extends` on lines 45 through 48](https://github.com/jshint/jshint/blob/12811e797a264f323b920b314db965b8adcb207e/tests/cli.js#L45-L48) for use in later tests, although a new test would probably be appropriate.

arikon commented:
@jugglinmike Done.

I've replaced `underscore` with `lodash` because:
- it is a bit faster
- it has `_.merge()` function for deep object extending (less code to extend configs and so to maintain)

And added two more tests.

rwaldron commented on package.json:
```diff
@@ -41,7 +41,7 @@
     "shelljs":             "0.3.x",
     "strip-json-comments": "1.0.x",
-    "underscore":          "1.8.x"
+    "lodash":              "3.6.x"
```
:D

rwaldron commented:
https://github.com/jshint/jshint/commit/928f19dbf3378734ef2119689dc51acfa37edfc2

arikon commented:
@rwaldron :+1:

arikon commented:
@rwaldron @jugglinmike Could you please `npm publish` `jshint@2.7.0`

jugglinmike commented:
All set
